Usage collector readme (#82548)
elastic/kibana: src/plugins/usage_collection/README.md

## Creating and Registering Usage Collector

Your usage collector needs to provide 
- a `type` for organizing your fields, 
- `schema` field to define the expected types of usage fields reported, 
- a `fetch` method for returning your usage data, and

> - an `isReady` method (that returns true or false) for letting the telemetry service know if it needs to wait for any asynchronous action. 

Reviewer: ```suggestion
- an `isReady` method (that returns true or false) for letting the telemetry service know if it needs to wait for any asynchronous action (initialization of clients or other services) before calling the `fetch` method. 
```

            total: 'long',
          },
        },

>         isReady: () => isCollectorFetchReady, // Method to return `true`/`false` to confirm if the collector is ready for the `fetch` method to be called.

Reviewer: NIT: It can be a promise/async method as well :)

Some background: 

- `MY_USAGE_TYPE` can be any string. It usually matches the plugin name. As a safety mechanism, we double check there are no duplicates at the moment of registering the collector.

> - `isReady` (added in v7.2.0 and v6.8.4) is a way for a usage collector to announce that some async process must finish first before it will return data synchronously (e.g. the task manager needs to run a task first). If any collector reports that it is not ready when we call its `fetch` method, we reset a flag to try again and, after a set amount of time, collect data from those collectors that are ready and skip any that are not. This means that if a collector returns `true` for `isReady` and it actually isn't ready to return data, there won't be telemetry data from that collector for the day. You should consider what it means if your collector doesn't return data in the first few documents when Kibana starts or, if we should wait for any other reason (e.g. the task manager needs to run your task first). If you need to tell telemetry collection to wait, you should implement this function with custom logic. If your `fetch` method will always return data synchronously or you're intentionally catching errors in your `fetch` method, then you can return true for `isReady` as shown in the example below.

Reviewer: ```suggestion
- `isReady` (added in v7.2.0 and v6.8.4) is a way for a usage collector to announce that some async process must finish first before it can return data in the `fetch` method (e.g. a client needs to be initialized, or the task manager needs to run a task first). If any collector reports that it is not ready when we call its `fetch` method, we reset a flag to try again and, after a set amount of time, collect data from those collectors that are ready and skip any that are not. This means that if a collector returns `true` for `isReady` and it actually isn't ready to return data, there won't be telemetry data from that collector in that telemetry report (usually once per day). You should consider what it means if your collector doesn't return data in the first few documents when Kibana starts or, if we should wait for any other reason (e.g. the task manager needs to run your task first). If you need to tell the telemetry collection to wait, you should implement this function with custom logic. If your `fetch` method can run without the need of any previous dependencies, then you can return true for `isReady` as shown in the example below.
```

- Changed the _return data synchronously_ because `fetch` can be async.
- Removed _or you're intentionally catching errors in your `fetch` method_ because we already catch the fetch method and exclude them from the report if they crash, without affecting others.

```ts
export const myCollector = makeUsageCollector<Usage>({
  type: 'my_working_collector',

>   isReady: () => true, // `fetch` doesn't call async methods

Reviewer: ```suggestion
  isReady: () => true, // `fetch` doesn't require any validation for dependencies to be met
```
